graphql2/graphqlapp/toolbox.go

```diff
@@ -0,0 +1,85 @@
+package graphqlapp
+
+import (
+	context "context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/target/goalert/config"
+	"github.com/target/goalert/graphql2"
+	"github.com/target/goalert/permission"
+	"github.com/ttacon/libphonenumber"
+)
+
+const twLookupURL = "https://lookups.twilio.com/v1/PhoneNumbers/"
+
+func (a *Mutation) DebugCarrierInfo(ctx context.Context, input graphql2.DebugCarrierInfoInput) (*graphql2.DebugCarrierInfo, error) {
+	// must be admin to fetch carrier info
+	err := permission.LimitCheckAny(ctx, permission.Admin)
+	if err != nil {
+		return nil, err
+	}
+
+	cfg := config.FromContext(ctx)
+	url := twLookupURL + input.Number + "?Type=carrier"
+	req, err := http.NewRequestWithContext(ctx, "GET", url, nil)
+	if err != nil {
+		return nil, err
+	}
+	req.SetBasicAuth(cfg.Twilio.AccountSID, cfg.Twilio.AuthToken)
+
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != 200 {
+		return nil, fmt.Errorf("non-200 response from Twilio: %s", resp.Status)
+	}
+
+	data, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	var result struct {
+		Carrier struct {
+			CC   string `json:"mobile_country_code"`
+			NC   string `json:"mobile_network_code"`
+			Name string
+			Type string
```

Comment:
do we need json tags here?

Reply:
No, the values that come in are `type` and `name` which will already parse correctly from the response, the others only needed it because of the `snake_case` is not handled by `UnmarshalJSON` (but upper/lower case is).